### ISSUE TITLE
Python classifier crash

### ISSUE THREAD

**Issue by forwardcjj** [BVLC/caffe#646]
F0708 13:14:07.033413 14991 pycaffe.cpp:116] Check failed: dims[0] == blob->num() (1 vs. 10) 
**\* Check failure stack trace: ***

I met this problem after successfully make && make test. 
but for make runtest. It keeps showing the curand not available skipping setting

Could someone offer some help to me? 

Thanks in advance. 

I0708 13:30:58.978436 15047 net.cpp:76] relu7 <- fc7
I0708 13:30:58.978476 15047 net.cpp:90] relu7 -> fc7 (in-place)
I0708 13:30:58.978505 15047 net.cpp:116] Top shape: 4096 1 1
I0708 13:30:58.978531 15047 net.cpp:137] relu7 needs backward computation.
I0708 13:30:58.978557 15047 net.cpp:66] Creating Layer drop7
I0708 13:30:58.978577 15047 net.cpp:76] drop7 <- fc7
I0708 13:30:58.978597 15047 net.cpp:90] drop7 -> fc7 (in-place)
I0708 13:30:58.978618 15047 net.cpp:116] Top shape: 4096 1 1
I0708 13:30:58.978634 15047 net.cpp:137] drop7 needs backward computation.
I0708 13:30:58.978656 15047 net.cpp:66] Creating Layer fc8
I0708 13:30:58.978674 15047 net.cpp:76] fc8 <- fc7
I0708 13:30:58.978693 15047 net.cpp:101] fc8 -> fc8
I0708 13:30:59.056764 15047 net.cpp:116] Top shape: 1000 1 1
I0708 13:30:59.056843 15047 net.cpp:137] fc8 needs backward computation.
I0708 13:30:59.056887 15047 net.cpp:66] Creating Layer prob
I0708 13:30:59.056916 15047 net.cpp:76] prob <- fc8
I0708 13:30:59.056942 15047 net.cpp:101] prob -> prob
I0708 13:30:59.056982 15047 net.cpp:116] Top shape: 1000 1 1
I0708 13:30:59.057005 15047 net.cpp:137] prob needs backward computation.
I0708 13:30:59.057024 15047 net.cpp:148] This network produces output prob
I0708 13:30:59.057046 15047 net.cpp:158] Collecting Learning Rate and Weight Decay.
I0708 13:30:59.057085 15047 net.cpp:152] Network initialization done.
F0708 13:31:01.359705 15047 pycaffe.cpp:116] Check failed: dims[0] == blob->num() (1 vs. 10) 
**\* Check failure stack trace: ***
Aborted

---

test script is simple 

net = imagenet.ImageNetClassifier(MODEL_FILE, PRETRAINED, True);
net.caffenet.set_phase_test();
net.caffenet.set_mode_cpu();
IMAGEFILE = 'dfasdf';
prediction = net.predict(IMAGE_FILE)
